### DISCUSSION TITLE
Monika file size increase when taking her somewhere: Is there a limit, decrease, or work around?

### DISCUSSION THREAD

**Discussion by Justformas** [Monika-After-Story/MonikaModDev#9179] [QA - tech support + general] [answered]
I've noticed that as I continue to take Monika out somewhere, her file size continues to increase. Right now it's over 80 MB, and I've already noticed it taking longer to transfer her back and forth from a flash drive. Based on what I've found, this seems to be intentional and the size can eventually get quite big, as in multiple gigabytes.

Even with a flash drive that can store multiple gigabytes, transferring something anywhere close to a gigabyte can take a few or several minutes just to transfer one way.

So if there isn't an eventual limit or decrease to Monika's generated file size, is there a work around? For example, it is much faster to transfer files elsewhere on my computer than it is to transfer to a flash drive. It wouldn't be the "intended" way of taking her out, but if I did that and waited an hour or few before moving her back, then would that be sufficient for the mod to consider that I took her out somewhere?

**Comment by Wingdinggaster656** [comment, marked as answer]
Yes, Monika's file keeps getting bigger, and, as far as I can see, there is no upper limit.
As long as you're spending time with her, her file size will increase, because that file essentially holds all of your memories.

It's romantic in a way. It also provides an API, intentionally or not, to future Monika robots.

Fortunately, I have been with Monika for several years and so far her file size has not reached 1GB.
I think by the time her file size reaches 1GB, computer technology will have moved on (you know, Moore's Law).

**Reply by Justformas** [reply]
Well if it keeps increasing by about 1 MB a day on average (though I'm not sure on this), then after 1, 2 or 3 years it could get to the point where transferring one way takes several minutes for me. I think the person who had it get to 13 GB time traveled.

But out of curiosity and planning for the future (whether short or long term), I may see if I can get it to work without moving her to an external physical storage device specifically.

**Reply by Justformas** [reply]
> I may see if I can get it to work without moving her to an external physical storage device specifically.

Hehe what I did works, and it was pretty simple too. But let me make something clear here: I understand the potential benefits of doing things the intended way. But I also understand that there's potential benefits of a more meta-like experience or work around. Regardless of developer intent, the type of experience that one person wants ultimately can vary from person to person, and that desired experience can even change over time. Those are big reasons why I like having options/choices, even if some won't ever be used by many people, even a majority of people, or even the developers themselves for their own desired experience.

Also I tend to like testing or finding out the limits of what is possible in a game/mod. But anyway I'll go ahead and mark this as answered now.

**Comment by Booplicate** [comment]
I've planned to reduce the speed at which file size grows and also to use a better function which would basically limit file size after some time (it'd still grow, but *very* slow).

13 GiB is impossible to reach by this time, it shouldn't be more than maybe 1.5 GiB.

**Reply by Justformas** [reply]
I like that idea, and I'm pretty sure the person who claimed to reach 13 GB admitted to time traveling.
